uber/automated_emails.py

```diff
@@ -17,14 +17,15 @@ class AutomatedEmail:
         Group: lambda session: session.query(Group).options(subqueryload(Group.attendees))
     }
 
-    def __init__(self, model, subject, template, filter, *, when=(),
+    def __init__(self, model, subject, template, filter, ident, *, when=(),
                  sender=None, extra_data=None, cc=None, bcc=None,
-                 post_con=False, needs_approval=True, ident=None, allow_during_con=False):
+                 post_con=False, needs_approval=True, allow_during_con=False):
 
         self.subject = subject.format(EVENT_NAME=c.EVENT_NAME)
-        self.ident = ident or self.subject
+        self.ident = ident
 
-        assert self.ident not in self.instances, 'error: ident "{}" is registered twice.'.format(self.ident)
+        assert self.ident, 'error: automated email ident may not be empty.'
+        assert self.ident not in self.instances, 'error: automated email ident "{}" is registered twice.'.format(self.ident)
```

Comment:
We no longer default to `subject` if `ident` is empty. Instead we assert that `ident` is not `Falsey` (not `None`, not an empty string, etc..). Correct me if I'm wrong, but I don't think we ever want to accept a `Falsey` `ident`.